ethcore/src/verification/queue/mod.rs

```diff
@@ -877,4 +887,57 @@ mod tests {
 		queue.collect_garbage();
 		assert_eq!(queue.num_verifiers(), 1);
 	}
+
+		#[test]
+		fn worker_threads_honor_specified_num_without_scaling() {
+			let spec = Spec::new_test();
+			let engine = spec.engine;
+			let mut config = Config::default();
+			config.verifier_settings.num_verifiers = 3;
+			config.verifier_settings.scale_verifiers = false;
+
+			let queue = BlockQueue::new(config, engine, IoChannel::disconnected(), true);
+
+			assert_eq!(queue.num_verifiers(), 3);
+		}
+
+		#[test]
+		fn worker_threads_specifyed_to_zero_should_set_to_one() {
+			let spec = Spec::new_test();
+			let engine = spec.engine;
+			let mut config = Config::default();
+			config.verifier_settings.num_verifiers = 1;
+			config.verifier_settings.scale_verifiers = false;
+
+			let queue = BlockQueue::new(config, engine, IoChannel::disconnected(), true);
+
+			assert_eq!(queue.num_verifiers(), 1);
+		}
+
+		#[test]
+		fn worker_threads_should_accept_max_eight() {
+			let spec = Spec::new_test();
+			let engine = spec.engine;
+			let mut config = Config::default();
+			config.verifier_settings.num_verifiers = 10000;
+			config.verifier_settings.scale_verifiers = false;
+
+			let queue = BlockQueue::new(config, engine, IoChannel::disconnected(), true);
+
+			assert_eq!(queue.num_verifiers(), 8);
+		}
+
+		#[test]
+		fn worker_threads_scaling_with_specifed_num_of_workers() {
+			let spec = Spec::new_test();
+			let engine = spec.engine;
+			let mut config = Config::default();
+			config.verifier_settings.num_verifiers = 5;
+			config.verifier_settings.scale_verifiers = true;
+
+			let queue = BlockQueue::new(config, engine, IoChannel::disconnected(), true);
+			queue.scale_verifiers(8);
+
+			assert_eq!(queue.num_verifiers(), 8);
```

Comment:
That's a good question and the answer is yes. I coded this because we assume that the maximum number of CPUs is 8 by the ```constant MAX_VERIFIERS``` but I think it doesn't make sense when thinking about it now! 

I will change this test and remove the constant!